### PR TITLE
Packaging improvements

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,11 +1,11 @@
 [metadata]
 name = shellcheck
-version = file: VERSION
 description = Python wrapper around invoking shellcheck (https://www.shellcheck.net/)
 long_description = file: README.md
 long_description_content_type = text/markdown
 url = https://github.com/ryanrhee/shellcheck-py
 author = Ryan Rhee
+author_email = ryan.y.rhee+pypi@gmail.com
```
Comment (asottile) [inline setup.cfg]: fixes

```
warning: Check: missing meta-data: if 'author' supplied, 'author_email' must be supplied too
```

File: setup.py [hunk 2]
```diff
@@ -1,107 +1,81 @@
...
+    return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
 
 
 def get_download_url() -> str:
-    return path.join(
-        'https://storage.googleapis.com/',
-        'shellcheck',
-        get_executable_name(),
-    )
+    return f'https://storage.googleapis.com/shellcheck/{get_executable_name()}'
```
Comment (asottile) [inline setup.py]: `os.path.join` isn't appropriate for url joining (on windows it'd put backslashes in there (!))

File: setup.py [hunk 3]
```diff
@@ -1,107 +1,81 @@
...
 
-from os import makedirs
-from os import fstat
-from os import fchmod
+from setuptools import setup
+from setuptools.command.install import install as orig_install
 
-from functools import lru_cache
+SHELLCHECK_VERSION = '0.7.0'
+PY_VERSION = '1'
```
Comment (asottile) [inline setup.py]: this is coupled with the `version=` keyword passed to `setup(...)` below, I arbitrarily picked `{shellcheck_version}.{py_version}` for a versioning scheme

File: setup.py [hunk 4]
```diff
@@ -1,107 +1,81 @@
...
 
 
 def get_arch() -> str:
-    if platform == "linux" or platform == "linux2":
+    if sys.platform.startswith('linux'):
```
Comment (asottile) [inline setup.py]: this is what's suggested by the docs, however python3 always reports as `'linux'` so we could just use that

File: setup.py [hunk 5]
```diff
@@ -1,107 +1,81 @@
...
         # TODO(rhee): detect "linux-aarch64" and "linux-armv6hf"
         return 'linux-x86_64'
-    elif platform == "darwin":
+    elif sys.platform == 'darwin':
         return 'darwin-x86_64'
-    elif platform == "win32" or platform == "win64":
+    elif sys.platform == 'win32':
```
Comment (asottile) [inline setup.py]: both 32 bit and 64 bit windows report as `'win32'`

File: setup.py [hunk 6]
```diff
@@ -1,107 +1,81 @@
...
 
 
 def download(url: str) -> bytes:
     with urlopen(url) as resp, urlopen(url + '.sha512sum') as shasum_resp:
-
         code = resp.getcode()
         scode = shasum_resp.getcode()
         if code != HTTPStatus.OK or scode != HTTPStatus.OK:
-            raise RuntimeError(f'HTTP failure. Codes: {code}, {scode}')
+            raise ValueError(f'HTTP failure. Codes: {code}, {scode}')
```
Comment (asottile) [inline setup.py]: I usually try and pick a more specific exception than `RuntimeError`

File: setup.py [hunk 7]
```diff
@@ -1,107 +1,81 @@
...
+            raise ValueError(
+                f'SHA512 mismatch! '
+                f'expected "{expected_shasum}" but got "{actual_shasum}"',
             )
 
         return data
 
 
 def save_executable(data: bytes, base_dir: str):
-    out_dir = path.join(base_dir, 'bin')
```
Comment (asottile) [inline setup.py]: windows is special snowflake and the directory is called `Scripts` there -- I corrected this below by using `install_scripts`

File: setup.py [hunk 8]
```diff
@@ -1,107 +1,81 @@
...
-    output_path = path.join(
-        out_dir,
-        'shellcheck',
-    )
-
-    makedirs(out_dir, exist_ok=True)
+    exe = 'shellcheck' if sys.platform != 'win32' else 'shellcheck.exe'
```
Comment (asottile) [inline setup.py]: we need to give the executable a proper extension on windows

File: setup.py [hunk 9]
```diff
@@ -1,107 +1,81 @@
...
-        # Mark as executable.
-        # https://stackoverflow.com/a/14105527
-        mode = fstat(fp.fileno()).st_mode
-        mode |= 0o111
-        fchmod(fp.fileno(), mode & 0o7777)
+    # Mark as executable.
+    # https://stackoverflow.com/a/14105527
+    mode = os.stat(output_path).st_mode
+    mode |= stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH
+    os.chmod(output_path, mode)
```
Comment (asottile) [inline setup.py]: chmod / stat are (iirc) more portable than fchmod / fstat (for windows)

File: setup.py [hunk 10]
```diff
@@ -148,27 +124,22 @@ def get_outputs(self):
...
-
-try:
-    from wheel.bdist_wheel import bdist_wheel as orig_bdist_wheel
-except ImportError:
-    pass
-else:
-    wheel_support()
-
-setup(cmdclass=command_overrides)
+setup(version=f'{SHELLCHECK_VERSION}.{PY_VERSION}', cmdclass=command_overrides)
```
Comment (asottile) [inline setup.py]: here's where the version is controlled -- when tagging the repo I'd use this version :+1:

File: setup.py (hunk 2 above)
Comment (ryanrhee) [reply setup.py]: oh derp, true

**Pull request merged**